### DISCUSSION TITLE
ST7789 (240x240 ST7789 GMT130-V1.0)

### DISCUSSION THREAD

**Discussion by fabse-hack** [lvgl-micropython/lvgl_micropython#148] [General]
Hello Kevin,

how i define the pins for an [240x240 ST7789 GMT130-V1.0](https://goldenmorninglcd.com/tft-display-module/1.3-inch-240x240-st7789-gmt130-v1.0/) Display?

The pinout is little bit confusied, because it have: GND, VCC, SCK, SDA, RES, DC, BLK

But - is not I2C, it is SPI.... - i read in the network, that i can set SDA to mosi, but where is miso ?
```python
GND =  ground
VCC = electricity 
SCK = is sck ....
SDA = is MOSI ?
RES = RESET
DC = is dc ...
BLK = electricity (background)
```
- with miso=None the firmware is not booting .

So, here is my code:

```python
spi_bus = SPI.Bus(host=1,
                  sck=1,
                  mosi=2
                  )

display_bus = lcd_bus.SPIBus(
    spi_bus=spi_bus,
    freq=40000000,
    dc=6
)

import st7789
import lvgl as lv


display = st7789.ST7789(
    data_bus=display_bus,
    display_width=240,
    display_height=240,
    reset_pin=4,
    reset_state=st7789.STATE_LOW,
    color_space=lv.COLOR_FORMAT.RGB565,
    rgb565_byte_swap=True
)

import task_handler
```

**Comment by kdschlosser** [comment]
it all depends on what pins you have the display connected to on the ESP32.

You want to use hardware SPI and Hardware I2C if possible. I would need to know what MCU you are using in order to tell you what pins to connect the display to.

**Reply by kdschlosser** [reply]
and if a pin is not in use it needs to not be set or you need to set it to -1

**Reply by fabse-hack** [reply]
ah thanks!

I have the ESP32 S2 mini.

- i will testing the -1 thanks

**Comment by fabse-hack** [comment]
i testing the same pins on the ESP32 S2 mini then my other project. But is not working. 

I think the display is not so good........

https://lupyuen.github.io/articles/st7789
so - better i order an ili9341 (then i know how it works :D)

**Comment by kdschlosser** [comment]
the S2 is going to have different pins to use.

lemme get you the pin numbers...

**Comment by kdschlosser** [comment]
SPI

MOSI - 13
MISO - 11
CS - 10
SCK - 12
HOST = 1


I2C
SCL = 17
SDA = 18
HOST = 1

**Reply by fabse-hack** [reply]
hm okay, thanks. I will test it!!
